src/ui/lp-render.ts

```diff
@@ -0,0 +1,236 @@
+                const field = tryOrPropogate(() => parseField(code));
+                if (!field.successful) {
+                    result = `Dataview (inline field '${code}'): ${field.error}`;
+                } else {
+                    const fieldValue = field.value;
+                    const intermediateResult = tryOrPropogate(() => executeInline(fieldValue, currentFile.path, index, dvSettings));
+                    if (!intermediateResult.successful) {
+                        result = `Dataview (for inline query '${fieldValue}'): ${intermediateResult.error}`;
+                    } else {
+                        const { value } = intermediateResult;
+                        result = value;
+                    }
+                }
+            } else if (dvSettings.inlineJsQueryPrefix.length > 0 && text.startsWith(dvSettings.inlineJsQueryPrefix)) {
+                if (dvSettings.enableInlineDataviewJs) {
+                    code = text.substring(dvSettings.inlineJsQueryPrefix.length).trim()
+                    try {
+                        // how do I set the `this` context properly?
+                        const comp = {
+                            api: api,
+                            current: () => currentFile,
+                            settings: dvSettings,
+                            index: index
+                        }
+                        if (code.includes("await")) {
+                            // await doesn't seem to work with it because the WidgetPlugin expects it to be synchronous
+                            // so this should be removed and instead an error message shown
+                            (evalWoContext("(async () => { " + PREAMBLE + code + " })()") as Promise<any>).then((value: any) => result = value)
+                        } else {
+                            result = evalWoContext(PREAMBLE + code);
+                        }
+
+                        function evalWoContext(script: string): any {
+                            return function () {return eval(script)}.call(comp);
+                        }
+                    } catch (e) {
+                        result = `Dataview (for inline JS query '${code}'): ${e}`;
+                    }
+                } else {
+                    result = "(disabled; enable in settings)";
+                }
+
+            } else {
+                return;
+            }
+
+            // at this point bounds contains the position we want to replace and
+            // result contains the text with which we want to replace it
+            const start = bounds.start;
+            const end = bounds.end;
+            if (selectionAndRangeOverlap(selection, start, end)) return;
+
+            widgets.push(
+                Decoration.replace({
+                    widget: new InlineWidget(result, dvSettings, currentFile),
+                    inclusive: false,
+                    block: false,
```

Comment:
If it's supposed to take more complex non-inline text like HTML for some values, we'd have to do the `Values` check before and not in the `toDom` method. There we could then either directly create the simple inline element (like now) or use the empty element line you propose and add to it later. We'd have to test if that works.

Reply:
It seems to work quite well with `span`s from my limited tests.